examples/routing_service/routing_service_file_adapter/cpp/FileInputDiscoveryStreamReader.cxx

```diff
@@ -0,0 +1,118 @@
+/*****************************************************************************/
+/*         (c) Copyright, Real-Time Innovations, All rights reserved.        */
+/*                                                                           */
+/*         Permission to modify and use for internal purposes granted.       */
+/* This software is provided "as is", without warranty, express or implied.  */
+/*                                                                           */
+/*****************************************************************************/
+
+#include "FileInputDiscoveryStreamReader.hpp"
+
+using namespace rti::routing;
+using namespace rti::routing::adapter;
+using namespace rti::community::examples;
+
+const std::string FileInputDiscoveryStreamReader::SQUARE_FILE_NAME =
+        "Input_Square.csv";
+const std::string FileInputDiscoveryStreamReader::CIRCLE_FILE_NAME =
+        "Input_Circle.csv";
+const std::string FileInputDiscoveryStreamReader::TRIANGLE_FILE_NAME =
+        "Input_Triangle.csv";
+
+bool FileInputDiscoveryStreamReader::fexists(const std::string filename)
+{
+    std::ifstream input_file;
+    input_file.open(filename);
+    return input_file.is_open();
+}
+
+FileInputDiscoveryStreamReader::FileInputDiscoveryStreamReader(
+        const PropertySet &,
+        StreamReaderListener *input_stream_discovery_listener)
+{
```

Comment:
The constructor is not exception safe. Any failure in push_back or new leaks the previously new'd StreamInfos.

You can make `data_samples_` a `vector<unique_ptr<StreamInfo>>`. That would also simplify the reutnr_loan function to just `v.clear()`.

Reply:
Done. We can't directly call clear(). I am using erase() with a range specified.